### ISSUE TITLE
Add AMM liquidity pool support

### ISSUE THREAD

**Issue by yxxyun** [ripple/rippled#3908]
## Summary
Add AMM liquidity pool to enchance XRPL DEX

## Motivation
It's hard to provide liquidity on current XRPL DEX, and some liquidity bots generate a lot noise offercreate and offercancel txs, this can be solved by AMM LP.

## Solution
Add AMM liquidity pool object can be deposit and withdraw, and update offercreate and path payment to try both liquidity pools and the order books.

## Paths Not Taken
Implement account based AMM LP via Hooks( Smart contract), but this can't trade against with offercreate and path payment.

**Comment by firefly3582** [comment]
This has been suggested many times over the years by different developers, but it seems like Ripple doesn't want to implement such thing, I don't know why.

*que the rothschild conspiracy theorist*

**Comment by StevieSimple** [comment]
Good point  I hope Ripple will do something to address a glaring shortfall? Peace Out Unity through Prosperity! LFBF TYHP

**Comment by yxxyun** [comment]
For a record : stellar AMM proposal https://github.com/stellar/stellar-protocol/blob/master/core/cap-0038.md

**Comment by nbougalis** [comment]
The team at Ripple is working on an AMM, which is something we've stated before.

With that said, let me take the opportunity to address something else: 

> This has been suggested many times over the years by different developers, but it seems like Ripple doesn't want to implement such thing, I don't know why.

I don't quite understand why it matters what Ripple does and doesn't want to implement. Anyone can propose changes and contribute code to effectuate those changes.

So, perhaps instead of waiting for Ripple to build X, whatever that X might be, how about writing up a spec and then implementing it instead?

**Comment by nbougalis** [comment]
@yxxyun: The C++ team at Ripple is working on an AMM proposal, which is something we've stated before.

---

With that said, let me take the opportunity to address something else. @firefly3582 wrote

>*This has been suggested many times over the years by different developers, but it seems like Ripple doesn't want to implement such thing, I don't know why.*

I don't quite understand why it matters what Ripple does and doesn't want to implement. Anyone can propose changes and contribute code to effectuate those changes.

So, perhaps instead of waiting for Ripple to build X, whatever that X might be, how about writing up a spec and then implementing it instead?

**Comment by develoQ** [comment]
https://github.com/XRPLF/rippled/pull/4294
